pycheribuild/projects/cross/linux.py

```diff
@@ -64,31 +65,31 @@ def check_system_dependencies(self) -> None:
         self.check_required_system_tool("dtc", apt="device-tree-compiler", homebrew="dtc")
 
     def _enable_config(self, option):
-        with open(self.source_dir / ".config") as f:
+        with open(self.build_dir / ".config") as f:
             lines = f.readlines()
         for i, line in enumerate(lines):
             if line.startswith(option + "=") or line.startswith("# " + option):
                 lines[i] = f"{option}=y\n"
                 break
         else:
             lines.append(f"{option}=y\n")
-        with open(self.source_dir / ".config", "w") as f:
+        with open(self.build_dir / ".config", "w") as f:
             f.writelines(lines)
 
     def setup(self) -> None:
         super().setup()
+        self.make_args.set(CROSS_COMPILE=str(self.CC.parent) + "/", LLVM=str(self.CC.parent) + "/")
         if self.crosscompile_target.is_riscv(include_purecap=True):
             self.linux_arch = "riscv"
-            self.make_args.set(CROSS_COMPILE="riscv64-unknown-elf-")
         elif self.crosscompile_target.is_aarch64(include_purecap=True):
             self.linux_arch = "arm64"
-            self.make_args.set(CROSS_COMPILE="aarch64-unknown-elf-")
 
         self.make_args.set(ARCH=self.linux_arch)
+        self.make_args.set(O=self.build_dir)
 
         # We only support building the kernel with LLVM/Clang
-        self.make_args.set(CC="clang")
-        self.make_args.set(LLVM="1")
```

Comment:
Does that work when you remove it? My understanding is that the Linux build system requires `LLVM=1` when building for Clang. https://docs.kernel.org/kbuild/llvm.html